### PR TITLE
[Ide] Dark theme support fixes for Mojave

### PR DESCRIPTION
**Pull request by sevoku**
Fixes VSTS #682823
Fixes VSTS #677025
Fixes VSTS #676679

### CODE REVIEW

**Comment by sevoku** [comment]
@monojenkins backport release-7.6

**Comment by sevoku** [comment]
@monojenkins backport release-7.7

File: main/src/addins/MacPlatform/MacPlatform.cs [hunk 1]
```diff
@@ -467,18 +467,20 @@ void InitApp (CommandManager commandManager)
...
-				var appearance = colorPanel.ContentView.Superview.Window.Appearance;
-				if (appearance == null)
-					appearance = NSAppearance.GetAppearance (IdeApp.Preferences.UserInterfaceTheme == Theme.Light ? NSAppearance.NameAqua : NSAppearance.NameVibrantDark);
-				// The subviews of the shared NSColorPanel do not inherit the appearance of the main panel window
-				// and need to be updated recursively.
-				UpdateColorPanelSubviewsAppearance (colorPanel.ContentView.Superview, appearance);
-			};
+			if (MacSystemInformation.OsVersion < MacSystemInformation.Mojave) { // the shared color panel has full automatic theme support on Mojave
+				Styles.Changed += (s, a) => {
+					var colorPanel = NSColorPanel.SharedColorPanel;
```
Comment (Therzok) [inline main/src/addins/MacPlatform/MacPlatform.cs]: We should be checking if `SharedColorPanelExists` is set.

File: main/src/addins/MacPlatform/MainToolbar/RunButton.cs [hunk 2]
```diff
@@ -68,7 +68,11 @@ void UpdateIcons (object sender = null, EventArgs e = null)
 
 		void UpdateCell ()
 		{
-			Appearance = NSAppearance.GetAppearance (IdeApp.Preferences.UserInterfaceTheme == Theme.Dark ? NSAppearance.NameVibrantDark : NSAppearance.NameAqua);
+			Appearance = IdeApp.Preferences.UserInterfaceTheme == Theme.Light
+				? NSAppearance.GetAppearance (NSAppearance.NameAqua)
+				: MacSystemInformation.OsVersion < MacSystemInformation.Mojave
+					? NSAppearance.GetAppearance (NSAppearance.NameVibrantDark)
+					: NSAppearance.GetAppearance (new NSString ("NSAppearanceNameDarkAqua"));
```
Comment (slluis) [inline main/src/addins/MacPlatform/MainToolbar/RunButton.cs]: This code is the same in 3 files. This logic could be implemented by a GetAppearance method in IdeTheme, and used from those 3 locations.

**Comment by sevoku** [comment]
@monojenkins backport release-7.7

**Comment by sevoku** [comment]
@monojenkins backport release-7.6

**Pull request merged**